youcompleteme does not build on latest unstable

## Basic info

To make sure that we are on the same page:

* Kernel: (run `uname -a`): `Linux yuu 4.3.6 #1-NixOS SMP Fri Feb 19 22:35:24 UTC 2016 x86_64 GNU/Linux`
* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...): `16.09pre78286.0bf8a1a (Emu)`
* Nix version: (run `nix-env --version`): `nix-env (Nix) 1.11.2`
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`): `"16.09pre78848.9549257"`

## Describe your issue here

`vimPlugins.youcompleteme` fails with

```
[...]
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/nix-build-vimplugin-youcompleteme-2016-03-08.drv-0/youcompleteme-381b213/build
make: *** No rule to make target 'ycm_support_libs'.  Stop.
builder for ‘/nix/store/vvwmin1545ibh9i89dq4dac8vpag9b61-vimplugin-youcompleteme-2016-03-08.drv’ failed with exit code 2
error: build of ‘/nix/store/vvwmin1545ibh9i89dq4dac8vpag9b61-vimplugin-youcompleteme-2016-03-08.drv’ failed
```

### Expected result

Build succeeds

### Actual result

Build fails

### Steps to reproduce

```
git checkout 954925771482b50493a24615c6e7e82e044a4fdf
nix-build -A vimPlugins.youcompleteme
```

### Maintainers

@marcweber @jagajaga

--- a/pkgs/misc/vim-plugins/default.nix
+++ b/pkgs/misc/vim-plugins/default.nix
@@ -883,11 +883,11 @@ rec {
   };
 
   youcompleteme = buildVimPluginFrom2Nix { # created by nix#NixDerivation
-    name = "youcompleteme-2016-03-08";
+    name = "youcompleteme-2016-03-10";
     src = fetchgit {
       url = "git://github.com/valloric/youcompleteme";
-      rev = "381b2132719a959f41e57ec0e6396fc1c3b6daf4";
-      sha256 = "1gski3s1960pmxisyq85awda0a3kb22ji9y76f67k1a4smy5q9xa";
+      rev = "f44435b88ec98156d17869aa67ad15f38cfecbf3";
+      sha256 = "1y50ilyfwj6rvpvg50iq418maxvsfs54i202v7x0lfs5hmvcb4hi";
     };
     dependencies = [];
     buildInputs = [
@@ -904,7 +904,7 @@ rec {
       mkdir build
       pushd build
       cmake -G "Unix Makefiles" . ../third_party/ycmd/cpp -DPYTHON_LIBRARIES:PATH=${python}/lib/libpython2.7.so -DPYTHON_INCLUDE_DIR:PATH=${python}/include/python2.7 -DUSE_CLANG_COMPLETER=ON -DUSE_SYSTEM_LIBCLANG=ON
-      make ycm_support_libs -j''${NIX_BUILD_CORES} -l''${NIX_BUILD_CORES}}
+      make ycm_core -j''${NIX_BUILD_CORES} -l''${NIX_BUILD_CORES}}
       ${python}/bin/python ../third_party/ycmd/build.py --gocode-completer --clang-completer --system-libclang
       popd
     '';

--- a/pkgs/misc/vim-plugins/vim2nix/additional-nix-code/youcompleteme
+++ b/pkgs/misc/vim-plugins/vim2nix/additional-nix-code/youcompleteme
@@ -12,7 +12,7 @@
       mkdir build
       pushd build
       cmake -G "Unix Makefiles" . ../third_party/ycmd/cpp -DPYTHON_LIBRARIES:PATH=${python}/lib/libpython2.7.so -DPYTHON_INCLUDE_DIR:PATH=${python}/include/python2.7 -DUSE_CLANG_COMPLETER=ON -DUSE_SYSTEM_LIBCLANG=ON
-      make ycm_support_libs -j''${NIX_BUILD_CORES} -l''${NIX_BUILD_CORES}}
+      make ycm_core -j''${NIX_BUILD_CORES} -l''${NIX_BUILD_CORES}}
       ${python}/bin/python ../third_party/ycmd/build.py --gocode-completer --clang-completer --system-libclang
       popd
     '';

